wavefront_python_sdk/common/utils.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import absolute_import, division, print_function
+
+import re
+import io
+from gzip import GzipFile
+import threading
+
+
+class AtomicCounter(object):
+    """
+    An atomic, thread-safe incrementing counter.
+    """
+
+    def __init__(self, initial=0):
+        self.value = initial
+        self._lock = threading.Lock()
+
+    def increment(self, num=1):
+        """
+        Increment atomic counter value
+        @param num: Num to be increased, 1 by default
+        @return: Current value after increment
+        """
+        with self._lock:
+            self.value += num
+            return self.value
+
+
+def chunks(data_list, batch_size):
+    """
+    Split list of data into chunks with fixed batch size
+    @param data_list: List of data
+    @param batch_size: Batch size of each chunk
+    @return: List of lists, [ [list of data, batch size length], [...] ... ]
+    """
+    """Yield successive batch_size chunks from l."""
+    for i in range(0, len(data_list), batch_size):
+        yield data_list[i:i + batch_size]
+
+
+def gzip_compress(data, compresslevel=9):
+    """
+    Compress data using GZIP
+    @param data: Data to compress
+    @param compresslevel: Compress Level
+    @return: Compressed data
+    """
+    buf = io.BytesIO()
+    with GzipFile(fileobj=buf, mode='wb', compresslevel=compresslevel) as f:
+        f.write(data)
+    return buf.getvalue()
+
+
+def sanitize(s):
+    """
+    Sanitize a string, replace whitespace with "-"
+    @param s: Input string
+    @return: Sanitized string
+    """
+    whitespace_sanitized = re.sub(r"[\s]+", "-", s)
```

Comment:
If the only purpose here is to replace the whitespaces, `str.replace(' ', '-')` would be way more efficient.

Reply:
\s includes not only ' ', but also '\t', '\n', etc